src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -186,7 +191,12 @@ public static string GetPathRoot(string path)
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
-            return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString : String.Empty;
+            return new string(GetPathRoot(path.AsReadOnlySpan()));
+        }
+
+        public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
+        {
+            return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString.AsReadOnlySpan() : ReadOnlySpan<char>.Empty;
```

Comment:
you missed this check that's in the string version: `if (PathInternal.IsEffectivelyEmpty(path))`
it needs to apply for the span version. if your tests are passing -- this means there is no test that `Path.GetPathRoot("    ")` throws. Please addo ne.